app/hocs/withTranslation.tsx

```diff
@@ -0,0 +1,17 @@
+import type { ComponentType } from 'react'
+import type { Language } from '../hooks/useLanguage'
+import { useLanguage } from '../hooks/useLanguage'
+
+export type EventualString = string | Promise<string> | ((...args: any[]) => (string | Promise<string>));
+export type Translation<T = Record<string, EventualString>> = Record<keyof T, EventualString>;
+export type Translations<T = Translation> = Record<Language, T>;
+export type PropsWithTranslation<T = Translation, P = unknown> = P & { translation: T };
```

Comment:
I'm uncertain what benefit a default value/type brings us here instead of constraining the input type.
I'd change this to `<T extends [X]>` where `[X]` is either `Record<string, EventualString>` or `Translation`.

Reply:
_Reopened  at https://github.com/helpwave/web/pull/14#discussion_r996277578_